### PR TITLE
Fixes to Grunt tasks; unifying GDI theme branding

### PR DESCRIPTION
**Pull request by anythingcodes**
Hi @Roenok & team -

This PR encourages a more consistent brand experience in our slides and fixes issues with the existing build tools / Grunt setup. It should also make editing easier and faster for instructors. Changes include: 

1. Fix SCSS compilation (in the existing configuration, no GDI theme SCSS files existed and our GDI theme files could be overwritten if Grunt was ran)
2. Upgrading Grunt and its dependencies
3. Adding BrowserSync for automatic updates (no refresh required)
4. Consistent brand experience across themes (font family, for example)
5. Include GDI themes only, to encourage a consistent brand experience and allow easier branching of GDI themes
6. Move editable and non-editable files into `src` and `dist` directories, respectively, to improve clarity

I'd also like to update our wiki with more details (similar to [this README](https://github.com/anythingcodes/gdi-es6-javascript#compiling-slides-and-running-locally)) about editing once these changes are merged.

Please give it a peek and let me know if you have any comments or suggestions. Thanks!

### CODE REVIEW

**Comment by Roenok** [comment]
@anythingcodes Hi Liz! I'll admit I am a bit confused about this PR. As it stands, you don't need to run any sort of build, Grunt or otherwise, to open and use the slides. You just download or pull the files and open index.html in any browser. Does this enhance the template so you can, optionally, create a local server? Or does it change it so it requires a local server?

**Comment by anythingcodes** [comment]
Hey @Roenok, thanks for reviewing! It's an enhancement so you can run a local server if you need (however, that won't impede anyone who wants to edit CSS only). Basically, our Grunt version and dependencies were out of date so you couldn't even run a local server if you wanted to with the last several releases of Node.js (which almost all developers are likely using at this point). There were also no compilation rules in place for GDI theme files.

Aside from that, the GDI themes were also inconsistent (different fonts, for example) so I fixed that. I also added some new features so that those who prefer running a local server can use BrowserSync so the tab with the localhost server open will automatically refresh whenever HTML/CSS/JS updates are made, which should make editing much faster for instructors. 

If you spin up this PR locally with Grunt and compare it to the current release, you can see what I mean about Grunt not running without an error, the Scss compilation not working, the occasional broken styles, the themes being different, etc.

**Tl;DR:** All in all, a ton of improvements to keep the branding consistent and fix the local server errors. However, nothing that would prevent someone from editing the CSS on its own if they don't want to run a local server. They can just open index.html and edit CSS files as you indicated.

Definitely let me know if you have any questions or suggestions, happy to help!

**Comment by Roenok** [comment]
@anythingcodes Thanks for the clarification! I'll go ahead and pull these changes in; it's a great bonus feature for instructors who use Grunt.
